dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -354,7 +363,10 @@ public void setDefaultBranchIfNotSet(Entry entry, String repositoryId) {
                             String reference = workflowVersion.getReference();
                             return branch.equals(reference);
                         }).findFirst();
-                firstWorkflowVersion.ifPresent(version -> entry.checkAndSetDefaultVersion(version.getName()));
+                firstWorkflowVersion.ifPresentOrElse(version -> entry.checkAndSetDefaultVersion(version.getName()), () -> {
+                    Version newestVersion = Collections.max(workflowVersions, Comparator.comparingInt(s -> s.getDbUpdateDate().getNanos()));
```

Comment:
Not sure this is the right date to sort on. This is generated by the DB with `@UpdateTimestamp`
I think this means that it will change when something like frozen is set.

Reply:
do you think lastBuilt for tags and lastModified for workflowversion is better?

Reply:
Looking at "Dates/Timestamps in Database for Tools and Workflows" in the shared drive, looks like yes